File: languages/java/src/main/java/com/bloock/sdk/Bloock.java
package com.bloock.sdk;

import com.bloock.sdk.bridge.proto.Config.NetworkConfig;
import com.bloock.sdk.entity.integrity.Network;
import java.util.HashMap;
import java.util.Map;

/**
 * Provides a centralized configuration for the Bloock SDK library. For information about Bloock SDK in Go, see https://bloock.com.
 */
public class Bloock {
  /**
   * Is a string variable representing the API key used for authentication with the Bloock SDK, create <a href="https://dashboard.bloock.com/login">here</a>.
   */
  public static String apiKey = "";
  /**
   * Is a string variable representing the host URL used for API communication with the Bloock SDK.
   */
  public static String apiHost = "";
  /**
   * Is a string variable representing the host URL used for Identity Managed API, required to be set for identity-related features of the Bloock SDK.
   */
  public static String identityApiHost = "";
  /**
   * Is a map variable that holds network configurations associated with specific network IDs in the Bloock SDK.
   */
  public static Map<Integer, NetworkConfig> networkConfig = new HashMap<>();

  /**
   * Sets the HTTP provider for the specified network in the Bloock SDK configuration.
   * @param network
   * @param provider
   */
  public static void setProvider(Network network, String provider) {
    if (Bloock.networkConfig.containsKey(network.toProto().ordinal())) {
      Bloock.networkConfig.put(
          network.toProto().ordinal(),
          Bloock.networkConfig.get(network.toProto().ordinal()).toBuilder()
              .setHttpProvider(provider)
              .build());
    } else {
      Bloock.networkConfig.put(
          network.toProto().ordinal(),
          NetworkConfig.newBuilder().setHttpProvider(provider).build());
    }
  }

  /**
   * Sets the contract address for the specified network in the Bloock SDK configuration.
   * @param network
   * @param contractAddress
   */
  public static void setContractAddress(Network network, String contractAddress) {
    if (Bloock.networkConfig.containsKey(network.toProto().ordinal())) {
      Bloock.networkConfig.put(
          network.toProto().ordinal(),
          Bloock.networkConfig.get(network.toProto().ordinal()).toBuilder()
              .setContractAddress(contractAddress)
              .build());
    } else {
      Bloock.networkConfig.put(
          network.toProto().ordinal(),
          NetworkConfig.newBuilder().setContractAddress(contractAddress).build());
    }
  }
}

File: languages/php/Bloock/Bloock.php
<?php

namespace Bloock;

/**
 * Provides a centralized configuration for the Bloock SDK library. For information about Bloock SDK in Go, see https://bloock.com.
 */
class Bloock
{
    /**
     * Is a string variable representing the API key used for authentication with the Bloock SDK, create [here](https://dashboard.bloock.com/login).
     * @var string
     */
    public static string $apiKey = "";
    /**
     * Is a string variable representing the host URL used for Identity Managed API, required to be set for identity-related features of the Bloock SDK.
     * @var string
     */
    public static string $identityApiHost = "";
    /**
     * Is a string variable representing the host URL used for API communication with the Bloock SDK.
     * @var string
     */
    public static string $apiHost = "";
    /**
     * Is an array variable that holds network configurations associated with specific network IDs in the Bloock SDK.
     * @var array
     */
    public static array $networkConfig = [];

    /**
     * Sets the HTTP provider for the specified network in the Bloock SDK configuration.
     * @param Network $network
     * @param string $provider
     * @return void
     */
    public static function setProvider(Network $network, string $provider)
    {
        if (Bloock::$networkConfig[Network::value($network)] != null) {
            Bloock::$networkConfig[Network::value($network)] =
                Bloock::$networkConfig[Network::value($network)]->setHttpProvider($provider);
        } else {
            $n = new NetworkConfig();
            $n->setHttpProvider($provider);
            Bloock::$networkConfig[Network::value($network)] = $n;
        }
    }

    /**
     * Sets the contract address for the specified network in the Bloock SDK configuration.
     * @param Network $network
     * @param string $contractAddress
     * @return void
     */
    public static function setContractAddress(Network $network, string $contractAddress)
    {
        if (Bloock::$networkConfig[Network::value($network)] != null) {
            Bloock::$networkConfig[Network::value($network)] =
                Bloock::$networkConfig[Network::value($network)]->setContractAddress($contractAddress);
        } else {
            $n = new NetworkConfig();
            $n->setContractAddress($contractAddress);
            Bloock::$networkConfig[Network::value($network)] = $n;
        }
    }
}